src/get-todo-config.ts

```diff
@@ -0,0 +1,82 @@
+import { join } from 'path';
+import { TodoConfig } from './types';
+
+/**
+ * Gets the todo configuration.
+ * Config values can be present in
+ *
+ * The package.json
+ *
+ * @example
+ * ```json
+ * {
+ *   "lintTodo": {
+ *     "daysToDecay": {
+ *       "warn": 5,
+ * 			 "error": 10
+ * 		 }
+ *   }
+ * }
+ * ```
+ *
+ * Environment variables (`TODO_DAYS_TO_WARN` or `TODO_DAYS_TO_ERROR`)
+ * 	- Env vars override package.json config
+ *
+ * Passed in directly, such as from command line options.
+ * 	- Passed in options override both env vars and package.json config
+ *
+ * @param baseDir - The base directory that contains the project's package.json.
+ * @param todoConfig - The optional todo configuration.
+ * @returns - The todo config object.
+ */
+export function getTodoConfig(
+  baseDir: string,
+  todoConfig: TodoConfig = {}
+): TodoConfig | undefined {
+  const daysToDecayPackageConfig = getFromPackageJson(baseDir);
+  const daysToDecayEnvVars = getFromEnvVars();
+  const mergedConfig = Object.assign({}, daysToDecayPackageConfig, daysToDecayEnvVars, todoConfig);
+
+  if (
+    typeof mergedConfig.warn === 'number' &&
+    typeof mergedConfig.error === 'number' &&
+    mergedConfig.warn >= mergedConfig.error
+  ) {
+    throw new Error(
+      'The `lintTodo` configuration in the package.json contains invalid values. The `warn` value must be less than the `error` value.'
+    );
+  }
+
+  return mergedConfig;
+}
+
+function getFromPackageJson(basePath: string): TodoConfig {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const pkg = require(join(basePath, 'package.json'));
```

Comment:
This will error if there is no package.json (which we don't want)

Reply:
Fixed.